service/service.go

```diff
@@ -0,0 +1,146 @@
+package service
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/ONSdigital/dp-reporter-client/reporter"
+	hierarchyHealthCheck "github.com/ONSdigital/go-ns/clients/hierarchy"
+	"github.com/ONSdigital/go-ns/elasticsearch"
+	"github.com/ONSdigital/go-ns/healthcheck"
+	"github.com/ONSdigital/go-ns/kafka"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/go-ns/rchttp"
+)
+
+// Service represents the necessary config for dp-dimension-extractor
+type Service struct {
+	EnvMax                   int64
+	BindAddr                 string
+	Consumer                 *kafka.ConsumerGroup
+	ElasticSearchURL         string
+	ErrorReporter            reporter.ImportErrorReporter
+	HealthcheckInterval      time.Duration
+	HealthcheckTimeout       time.Duration
+	HierarchyAPIURL          string
+	HTTPClient               *rchttp.Client
+	MaxRetries               int
+	SearchBuiltProducer      kafka.Producer
+	SearchBuilderErrProducer kafka.Producer
+	SearchBuilderURL         string
+	Shutdown                 time.Duration
+}
+
+// Start handles consumption of events
+func (svc *Service) Start() {
+	log.Info("application started", log.Data{"search_builder_url": svc.SearchBuilderURL})
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+
+	eventLoopDone := make(chan bool)
+	errorChannel := make(chan error, 1)
+
+	svc.HTTPClient = rchttp.DefaultClient
+
+	eventLoopContext, eventLoopCancel := context.WithCancel(context.Background())
+
+	healthChecker := healthcheck.NewServer(
+		svc.BindAddr,
+		svc.HealthcheckInterval,
+		errorChannel,
+		hierarchyHealthCheck.New(svc.HierarchyAPIURL),
+		elasticsearch.NewHealthCheckClient(svc.ElasticSearchURL),
+	)
+
+	// eventLoop
+	go func() {
+		defer close(eventLoopDone)
+		for {
+			select {
+			case <-eventLoopContext.Done():
+				log.Trace("Event loop context done", log.Data{"eventLoopContextErr": eventLoopContext.Err()})
+				return
+			case message := <-svc.Consumer.Incoming():
+
+				instanceID, dimension, err := svc.handleMessage(eventLoopContext, message)
```

Comment:
probably worth using `logData := log.Data{"func": "service.Start.eventLoop", "instance_id": instanceID, "dimension": dimension, "kafka_offset": message.Offset()}` here (and in all log lines below)